src/app/delegation_verify/submission.ml

```diff
@@ -0,0 +1,276 @@
+      : submission )
+
+  let created_at ({ created_at; _ } : submission) = created_at
+
+  let block_hash ({ block_hash; _ } : submission) = block_hash
+
+  let snark_work ({ snark_work; _ } : submission) = snark_work
+
+  let submitter ({ submitter; _ } : submission) = submitter
+
+  let load_submissions { conf ; period_start; period_end } =
+    let open Deferred.Or_error.Let_syntax in
+    let start_day = Time.of_string period_start |> Time.to_date ~zone:Time.Zone.utc in
+    let end_day = Time.of_string period_end |> Time.to_date ~zone:Time.Zone.utc in
+    let partition_keys =
+      Date.dates_between ~min:start_day ~max:end_day
+      |> List.map ~f:(fun d -> Date.format d "%Y-%m-%d")
+    in
+    let partition =
+      if List.length partition_keys = 1 then
+        sprintf "submitted_at_date = '%s'" (List.hd_exn partition_keys)
+      else
+        sprintf
+          "submitted_at_date IN (%s)"
+          (String.concat ~sep:"," @@ List.map ~f:(sprintf "'%s'") partition_keys)
+    in
+    let%bind raw =
+      Cassandra.select ~conf ~parse:raw_of_yojson
+        ~fields:
+          [ "created_at"
+          ; "submitted_at_date"
+          ; "submitted_at"
+          ; "peer_id"
+          ; "snark_work"
+          ; "remote_addr"
+          ; "submitter"
+          ; "block_hash"
+          ; "graphql_control_port"
+          ]
+        ~where:
+          (sprintf
+             "%s AND submitted_at >= '%s' AND submitted_at <= '%s'"
+             partition
+             period_start period_end )
+        "submissions"
+    in
+    List.fold_right raw ~init:(Ok []) ~f:(fun sub acc ->
+        let open Result.Let_syntax in
+        let%bind l = acc in
+        let snark_work =
+          Option.map sub.snark_work ~f:(fun s ->
+              String.chop_prefix_exn s ~prefix:"0x"
+              |> Hex.Safe.of_hex |> Option.value_exn |> Base64.encode_string )
+        in
+        let%map s = of_raw { sub with snark_work } in
+        s :: l )
+    |> Deferred.return
+
+  let load_block ~block_hash { conf; _ } =
+    let open Deferred.Or_error.Let_syntax in
+    let%bind block_data =
+      Cassandra.select ~conf ~parse:block_data_of_yojson
+        ~fields:[ "raw_block" ]
+        ~where:(sprintf "block_hash = '%s'" block_hash)
+        "blocks"
+    in
+    match List.hd block_data with
+    | None ->
+        Deferred.Or_error.error_string "Cassandra: Block not found"
+    | Some b ->
+        String.chop_prefix_exn b.raw_block ~prefix:"0x"
+        |> Hex.Safe.of_hex |> Option.value_exn |> return
+
+  let output { conf; _ } (submission : submission) = function
+    | Ok payload ->
+        Output.display payload ;
+        Cassandra.update ~conf
+          ~table:"submissions"
+          ~where:
+            (sprintf
+               "submitted_at_date = '%s' and submitted_at = '%s' and submitter \
+                = '%s'"
+               (List.hd_exn @@ String.split ~on:' ' submission.created_at)
+               submission.created_at
```

Comment:
Ah, I think this may be the culprit of the issue I was seeing...

```suggestion
               (List.hd_exn @@ String.split ~on:' ' submission.submitted_at)
               submission.submitted_at
```
Since submitted_at and created_at are not the same (one is server date and other is request date)... updating the row may lead (in the circumstance when they are different (mostly they are not)) to adding additional row with partial nulls.